GooglePlayInstant/Editor/PluginUpdateWindow.cs

```diff
@@ -0,0 +1,339 @@
+                HandleVersionCheckFailed("Latest release assets are empty.", responseText);
+                return;
+            }
+
+            if (latestReleaseAssets.Length > 1)
+            {
+                HandleVersionCheckFailed(string.Format(
+                    "Latest release unexpectedly has {0} asset files.", latestReleaseAssets.Length), responseText);
+                return;
+            }
+
+            var latestReleaseAsset = latestReleaseAssets[0];
+            if (latestReleaseAsset == null)
+            {
+                HandleVersionCheckFailed("Latest release asset is null.", responseText);
+                return;
+            }
+
+            if (string.IsNullOrEmpty(latestReleaseAsset.browser_download_url))
+            {
+                HandleVersionCheckFailed("Latest release download URL is missing.", responseText);
+                return;
+            }
+
+            _latestReleaseVersion = latestReleaseResponse.tag_name;
+            _latestReleaseDownloadUrl = latestReleaseAsset.browser_download_url;
+
+            Debug.LogFormat("Plugin version check result: current={0} latest={1} url={2}",
+                GooglePlayInstantUtils.PluginVersion, _latestReleaseVersion, _latestReleaseDownloadUrl);
+        }
+
+        private void HandleVersionCheckFailed(string versionCheckError, string additionalErrorText)
+        {
+            if (additionalErrorText == null)
+            {
+                Debug.LogError(versionCheckError);
+            }
+            else
+            {
+                Debug.LogErrorFormat("{0}: {1}", versionCheckError, additionalErrorText);
+            }
+
+            var message = string.Format("{0}\n\nClick \"{1}\" to retry.", versionCheckError, CheckReleasesButtonText);
+            EditorUtility.DisplayDialog("Version Check Failed", message, WindowUtils.OkButtonText);
+        }
+
+        private void HandlePluginDownloadRequestIsDone()
+        {
+            if (GooglePlayInstantUtils.IsNetworkError(_pluginDownloadRequest))
+            {
+                var downloadRequestError = _pluginDownloadRequest.error;
+                Debug.LogErrorFormat("Plugin download error: {0}", downloadRequestError);
+                EditorUtility.DisplayDialog("Download Failed", downloadRequestError, WindowUtils.OkButtonText);
+                return;
+            }
+
+            GooglePlayInstantUtils.FinishFileDownload(_pluginDownloadRequest, _latestReleaseSavePath);
+
+            Debug.LogFormat("Plugin downloaded: {0}", _latestReleaseSavePath);
+            if (_shouldInstallPlugin)
+            {
+                AssetDatabase.ImportPackage(_latestReleaseSavePath, true);
+            }
+            else
+            {
+                var message = string.Format("The plugin has been downloaded: {0}", _latestReleaseSavePath);
+                if (EditorUtility.DisplayDialog("Download Complete", message, WindowUtils.OkButtonText))
+                {
+                    Close();
```

Comment:
This throws a null ref exception because it destroys the window in the middle of its Update loop. Replacing it with                     `EditorApplication.delayCall += Close;` fixes it.

Reply:
Done here and in BundletoolDownloadWindow.